config/retry/config.go

```diff
@@ -96,6 +98,50 @@ func NewConfig(name string, metric *prometheus.Observer, backoffFnCfg *BackoffFn
 	}
 }
 
+type RetryRateLimiter struct {
+	tokenCount                 int32
+	allowedRetryToSuccessRatio float32
+	cap                        int32
+}
+
+func NewRetryRateLimiter(cap int32, ratio float32) *RetryRateLimiter {
```

Comment:
Ditto for the comments, and better to explain the meaning of the input parameters.

Besides, would it be less expensive using `int` or `uint` type for `ratio` instead of float values?